src/components/Pokedex/PokeCard/PokeCard.tsx

```diff
@@ -1,13 +1,47 @@
-import React from "react";
-import "./PokeCard.css";
-
-const PokeCard = (props: any) => {
-    return (
-        <div className="pokecard">
-            <img src={props.url} alt="Default for Pokemon"/>
-            <p>{props.name}</p>
-        </div>
-    )
+// @ts-nocheck
+import React, { useEffect, useState } from 'react';
+import './PokeCard.css';
+
+function useSinglePokemonDetails(pokemon: any) {
+  const [details, setDetails] = useState(null);
+
+  useEffect(
+    function () {
+      async function fetchDetails() {
+        try {
+          const pokemonDetailsResponse = await fetch(pokemon.url);
+
+          if (!pokemonDetailsResponse.ok) {
+            throw Error(`${response.status}: ${response.statusText}`);
+          }
+
+          setDetails(await pokemonDetailsResponse.json());
+        } catch (error) {
+          console.log(error);
+        }
+      }
+
+      fetchDetails();
+    },
+    [pokemon.url]
```

Comment:
I really like putting the url that returns the data for each card in the dependency array.  Brilliant move -- and I think it will really help performance as fewer cards will be rendered at a time.